src/QirRuntime/test/Standalone/InputTypes/.clang-tidy

```diff
@@ -0,0 +1,3 @@
+Checks: 
```

Comment:
why do you need a special .clang-tidy file in this folder? why inheriting it from the parent folders isn't OK?

Reply:
The main reason was to change `HeaderFilterRegex` to exclude `CLI11.hpp` since it's a third-party header that makes clang-tidy unhappy. I haven't gotten to do this specifically yet but do you have any pointer on how to do that? So far, I haven't found examples on how to do this in the clang-tidy documentation.

Reply:
I don't like this. I don't like pulling into production code a foreign header, I like it even less that this header forces us to suppress static analysis... If we _must_ use the header, we might be able to suppress clang-tidy for it selectively with:
```
  --header-filter=<string>       -
                                   Regular expression matching the names of the
                                   headers to output diagnostics from. Diagnostics
                                   from the main file of each translation unit are
                                   always displayed.
                                   Can be used together with -line-filter.
                                   This option overrides the 'HeaderFilterRegex'
                                   option in .clang-tidy file, if any.
  --line-filter=<string>         -
                                   List of files with line ranges to filter the
                                   warnings. Can be used together with
                                   -header-filter. The format of the list is a
                                   JSON array of objects:
                                     [
                                       {"name":"file1.cpp","lines":[[1,3],[5,7]]},
                                       {"name":"file2.h"}
                                     ]
```
per https://clang.llvm.org/extra/clang-tidy/ (but I haven't tried it)

Reply:
I added `InheritParentConfig: true` to the `.clang-tidy` file for the standalone samples, it is an opt-in check for that directory.

Ideally, `HeaderFilterRegex` should have a negative lookahead regex to exclude particular files and include the rest but this is not currently supported. There has been multiple attempts to do this ([here](https://reviews.llvm.org/D34654) and [here](http://clang-developers.42468.n3.nabble.com/clang-tidy-Negative-Lookahead-Support-td4048783.html)) but its seems like they got no traction.

The other option was `--line-filter` but that is not supported as a configuration file option either.